jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/editor/S3TreeTable.kt

```diff
@@ -67,12 +74,46 @@ class S3TreeTable(
     private val mouseListener = object : MouseAdapter() {
         override fun mouseClicked(e: MouseEvent) {
             val row = rowAtPoint(e.point).takeIf { it >= 0 } ?: return
+            handleOpeningFile(row, e)
             handleLoadingMore(row, e)
         }
     }
 
+    private fun handleOpeningFile(row: Int, e: MouseEvent) {
+        if (e.clickCount < 2) {
+            return
+        }
+        val objectNode = (tree.getPathForRow(row).lastPathComponent as? DefaultMutableTreeNode)?.userObject as? S3TreeObjectNode ?: return
+        if (objectNode.size > S3TreeObjectNode.MAX_FILE_SIZE_TO_OPEN_IN_IDE) {
```

Comment:
This should probably use the IDE standards:

com.intellij.openapi.util.io.FileUtilRt#LARGE_FOR_CONTENT_LOADING